department-of-veterans-affairs/caseflow
# Allow editing document ID

**Why**: Sometimes, attorneys enter the wrong document ID. They need to
be able to correct the ID after the fact.

**How**:
For now, we only display the Edit option for AMA cases.
After the user enters the new document ID, the backend performs two
validations:
- The current user must be either the assigned attorney or judge.
Although all users can view any appeal, only those who are assigned to
it are able to change the document ID.
- The document ID must follow a specific format:
12345-12345678 or 12345678.123 or 12345678.1234 for Draft Decisions

Resolves #7606

Testing Plan:
- View queue as user BVASCASPER1
- Click on any AMA case and complete the checkout flow
- Go back to the appeal details page
You should now see an "Edit" button next to the Document ID
- Click on the Edit button
You should see a modal with a text field pre-filled with the document
ID. The Save button should be disabled by default.
- Start typing in the field
The Save button should now be enabled.
- Enter an invalid ID, such as "1234" and click Save
You should see an error message in the modal letting you know which
formats are valid.
- Enter a valid ID, such as "12345678.123" and click Save
The modal should be dismissed, and the new document ID should appear on
the details page
